### ISSUE TITLE
Replace fork with transforms

### ISSUE THREAD

**Issue by nearmax** [nearprotocol/assemblyscript#74]
AssemblyScript fork requires constant maintenance and periodic fires in our system. Interestingly, the only purpose of the fork is to be able to inject the code into our system by wrapping methods into JSON serialization and deserialization code.

Luckily, AssemblyScript supports transforms https://docs.assemblyscript.org/details/peculiarities#transforms which can alter AssemblyScript before and after parsing on AST level. Upon reading the small bits of documentation I do not see the reason why it cannot be used to achieve our goal without forking the compiler. (For instance, in Rust we are using procedural macros to generate the very same JSON serialization/deserialization wrapping code, and procedural macros work on AST level).

We should either:
* Implement transforms that inject our code and delete the fork;
* Find an example of the code that we want to inject that cannot be injected with transforms.

**Comment by bowenwang1996** [comment]
Right now it all depends on https://github.com/AssemblyScript/assemblyscript/pull/866

**Comment by willemneal** [comment]
Yeah there are still a few other sticking points that we need to keep the fork.  For example, I've added a check to make sure floating point numbers are not used.  I think the major issue has been the lack of versioning on our side and AS's side, which meant changes to master affected everything.

**Comment by nearmax** [comment]
If the floating point check is the only things that prevents us from getting rid of the fork then let's remove this check. We already check for floating points before contract execution. Anything else that prevents us from getting rid of the fork?
